### ISSUE TITLE
gnome 40 compatibility

### ISSUE THREAD

**Issue by SamLukeYes** [SebastianLuebke/improved-osk-gnome-ext#13]
Recently, Arch Linux has upgraded `gnome-shell` to 40.0. Here are some issues I've found so far (tested on Wayland with the latest git version of this extension):
1. ~~Need to set `/org/gnome/shell/disable-extension-version-validation` to `true` in dconf to enable this extension~~
2. OSK cannot show up when a physical mouse is available (my mouse is connected through bluetooth)
3. Application windows cannot be resized when OSK shows up
4. The extension settings can't be displayed (technical details pasted below)
```
Error: No property margin on GtkGrid

Stack trace:
  _init/Gtk.Widget.prototype._init@resource:///org/gnome/gjs/modules/core/overrides/Gtk.js:45:40
  buildPrefsWidget@/usr/share/gnome-shell/extensions/improvedosk@luebke.io/prefs.js:25:21
  _init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:209:40
  OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
  asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:115:22
  run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:177:20
  main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
  run@resource:///org/gnome/gjs/modules/script/package.js:206:19
  start@resource:///org/gnome/gjs/modules/script/package.js:190:8
  @/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
  ```

**Comment by luebke-dev** [comment]
Yeah, feel free to submit a pull request. got a new notebook without touchscreen so I don't  need this extension anymore. Am 09.04.2021 05:59 schrieb SamLukeYes ***@***.***>:
Recently, Arch Linux has upgraded gnome-shell to 40.0. Here are some issues I've found so far (tested on Wayland with the latest git version of this extension):
Need to set /org/gnome/shell/disable-extension-version-validation to true in dconf to enable this extensionOSK cannot show up when a physical mouse is available (my mouse is connected through bluetooth)Application windows cannot be resized when OSK shows upThe extension settings can't be displayed (technical details pasted below)
Error: No property margin on GtkGrid

Stack trace:
  ***@***.***:///org/gnome/gjs/modules/core/overrides/Gtk.js:45:40
  buildPrefsWidget@***@***.***/prefs.js:25:21
  ***@***.***:///org/gnome/Shell/Extensions/js/extensionsService.js:209:40
  ***@***.***:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
  ***@***.***:///org/gnome/gjs/modules/core/overrides/Gio.js:115:22
  ***@***.***:///org/gnome/Shell/Extensions/js/dbusService.js:177:20
  ***@***.***:///org/gnome/Shell/Extensions/js/main.js:19:13
  ***@***.***:///org/gnome/gjs/modules/script/package.js:206:19
  ***@***.***:///org/gnome/gjs/modules/script/package.js:190:8
  @/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
